mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,646 @@
+
+    return metrics
+
+
+def _get_classifier_per_class_metrics_collection_df(y, y_pred, labels):
+    per_class_metrics_list = []
+    for positive_class_index, positive_class in enumerate(labels):
+        (y_bin, y_pred_bin, _,) = _get_binary_sum_up_label_pred_prob(
+            positive_class_index, positive_class, y, y_pred, None
+        )
+
+        per_class_metrics = {"positive_class": positive_class}
+        per_class_metrics.update(_get_classifier_per_class_metrics(y_bin, y_pred_bin))
+        per_class_metrics_list.append(per_class_metrics)
+
+    return pd.DataFrame(per_class_metrics_list)
+
+
+_Curve = namedtuple("_Curve", ["plot_fn", "plot_fn_args", "auc"])
+
+
+def _gen_classifier_curve(
+    is_binomial,
+    y,
+    y_probs,
+    labels,
+    curve_type,
+):
+    if curve_type == "roc":
+
+        def gen_x_y_thresholds_fn(_y, _y_prob):
+            fpr, tpr, _thresholds = sk_metrics.roc_curve(_y, _y_prob)
+            return fpr, tpr, _thresholds
+
+        xlabel = "False Positive Rate"
+        ylabel = "True Positive Rate"
+        legend_loc = "lower right"
+    elif curve_type == "pr":
+
+        def gen_x_y_thresholds_fn(_y, _y_prob):
+            precision, recall, _thresholds = sk_metrics.precision_recall_curve(_y, _y_prob)
+            _thresholds = np.append(_thresholds, [1.0], axis=0)
+            return recall, precision, _thresholds
+
+        xlabel = "recall"
+        ylabel = "precision"
+        legend_loc = "lower left"
+    else:
+        assert False, "illegal curve type"
+
+    if is_binomial:
+        x_data, y_data, thresholds = gen_x_y_thresholds_fn(y, y_probs)
+        data_series = [("positive class", x_data, y_data)]
+        legend_loc = None
+        auc = sk_metrics.auc(x_data, y_data)
+    else:
+        curve_list = []
+        for positive_class_index, positive_class in enumerate(labels):
+            y_bin, _, y_prob_bin = _get_binary_sum_up_label_pred_prob(
+                positive_class_index, positive_class, y, None, y_probs
+            )
+
+            x_data, y_data, thresholds = gen_x_y_thresholds_fn(y_bin, y_prob_bin)
+            curve_list.append((positive_class, x_data, y_data, thresholds))
+
+        data_series = [
+            (f"Positive Class = {positive_class}", x_data, y_data)
+            for positive_class, x_data, y_data, _ in curve_list
+        ]
+        auc = [sk_metrics.auc(x_data, y_data) for _, x_data, y_data, _ in curve_list]
+    return _Curve(
+        plot_fn=plot_lines,
+        plot_fn_args={
+            "data_series": data_series,
+            "xlabel": xlabel,
+            "ylabel": ylabel,
+            "legend_loc": legend_loc,
+            "line_kwargs": {"drawstyle": "steps-post"},
+        },
+        auc=auc,
+    )
+
+
+_matplotlib_initialized = False
+
+
+def _init_matplotlib():
+    global _matplotlib_initialized
+    if not _matplotlib_initialized:
+        import matplotlib.pyplot as pyplot
+
+        pyplot.rcParams["figure.dpi"] = 144
```

Comment:
I don't think we should mutate `rcParams`. Can we just specify dpi and figsize when saving plots?

Reply:
maybe we could try this
https://stackoverflow.com/a/35394637

Reply:
Yes. I just also found it. Use `with matplotlib.rc_context(...)`